examples/argo.py

```diff
@@ -0,0 +1,230 @@
+"""
+ARGO: An Auto-Tuning Runtime System for Scalable GNN Training on Multi-Core Processor
+--------------------------------------------
+Graph Neural Network (GNN) training suffers from low scalability on multi-core CPUs. 
+Specificially, the performance often caps at 16 cores, and no improvement is observed when applying more than 16 cores.
+ARGO is a runtime system that offers scalable performance by overlapping the computation and communication during GNN training.
+With ARGO enabled, we are able to scale over 64 cores, allowing ARGO to speedup GNN training (in terms of epoch time) by up to 5.06x and 4.54x on a Xeon 8380H and a Xeon 6430L, respectively.
+--------------------------------------------
+Paper Link: https://arxiv.org/abs/2402.03671
+"""
+
+import time
+from typing import Callable, Tuple
+
+import dgl.multiprocessing as dmp
+import psutil
+from skopt import gp_minimize
+
+
+class ARGO:
+
+    def __init__(
+        self,
+        n_search=10,
+        epoch=200,
+        batch_size=4096,
+        space=[(2, 8), (1, 4), (1, 32)],
+        random_state=1,
+    ):
+        """
+        Initialization
+
+        Parameters
+        ----------
+        n_search: int
+            Number of configuration searches the auto-tuner will conduct
+
+        epoch: int
+            Number of epochs of GNN training
+
+        batch_size: int
+            Size of the mini-batch
+
+        space: list[Tuple(int,int)]
+            Range of the search space; [range of processes, range of samplers for each process, range of trainers for each process]
+
+        random_state: int
+            Number of random initializations before searching
+
+        acq_function: str
```

Comment:
Looks like `acq_function` isn't used. Mind updating the docsting or the code?

Reply:
Sure thing.